src/os/bluestore/BlueStore.cc

```diff
@@ -3373,14 +3373,21 @@ bool BlueStore::exists(CollectionHandle &c_, const ghobject_t& oid)
   dout(10) << __func__ << " " << c->cid << " " << oid << dendl;
   if (!c->exists)
     return false;
-  RWLock::RLocker l(c->lock);
-  OnodeRef o = c->get_onode(oid, false);
+
+  bool r = true;
+
+  {
+    RWLock::RLocker l(c->lock);
+    OnodeRef o = c->get_onode(oid, false);
+    if (!o || !o->exists)
+      r = false;
+  }
+
```

Comment:
Negative, `cache->trim()` does cache cleanup only and is under its own mutex protection already.



> We can't do that cache->trim needs to be within collection lock as it is modifying onode structures..

Otherwise shouldn't we promote to `WLocker(c->lock)` here?